test/parallel/test-http2-client-window-size.js

```diff
@@ -0,0 +1,104 @@
+'use strict';
+
+// This test ensures that servers are able to send data independent of window
+// size.
+
+const assert = require('assert');
+const common = require('../common');
+const h2 = require('http2');
+
+// Helper method which takes an array of buffers and returns a buffer containing
+// all the bytes in each buffer in order
+function concatBuffers(buffers) {
+  const resultLength =
+    buffers.reduce((prev, buf) => prev + buf.length, 0);
+  const result = Buffer.alloc(resultLength);
+  let offset = 0;
+  for (let i = 0; i < buffers.length; i++) {
+    buffers[i].copy(result, offset);
+    offset += buffers[i].length;
+  }
+  return result;
+}
+
+// Given a list of buffers and an initial window size, have a server write
+// each buffer to the HTTP2 Writable stream, and let the client verify that
+// all of the bytes were sent correctly
+function run(buffers, initialWindowSize) {
+  const expectedBuffer = concatBuffers(buffers);
+
+  const server = h2.createServer();
+  const a = [];
+  server.on('stream', (stream, headers, flags) => {
+    let p = Promise.resolve();
+    for (let i = 0; i < buffers.length; i++) {
+      p = p.then(() => new Promise(resolve => {
+        const drained = stream.write(buffers[i], 'buffer', resolve);
+        if (!drained) {
+          stream.once('drain', resolve);
+        }
+      }));
+    }
+    p = p.then(() => {
+      stream.end();
+    });
+  });
```

Comment:
this block is not really idiomatic stream code, also it allocates a lot of functions and promises that are not needed. Within a function, you can iterate over `buffers` as long as `write` returns `true`, when `false`, you can setup a `once('drain')` handler to restart from where it stopped.